kernel/src/debug.rs

```diff
@@ -527,23 +527,45 @@ impl Write for DebugWriterWrapper {
     }
 }
 
-pub fn begin_debug_fmt(args: Arguments) {
+pub fn debug_format(args: Arguments) {
     let writer = unsafe { get_debug_writer() };
 
     let _ = write(writer, args);
-    let _ = writer.write_str("\r\n");
     writer.publish_bytes();
 }
 
-pub fn begin_debug_verbose_fmt(args: Arguments, file_line: &(&'static str, u32)) {
+pub fn debug_formatln(args: Arguments) {
     let writer = unsafe { get_debug_writer() };
 
+    let _ = write(writer, args);
+    let _ = writer.write_str("\r\n");
+    writer.publish_bytes();
+}
+
+fn print_debug_header(
+    writer: &mut DebugWriterWrapper,
+    args: Arguments,
+    file_line: &(&'static str, u32),
+) {
     writer.increment_count();
     let count = writer.get_count();
 
     let (file, line) = *file_line;
     let _ = writer.write_fmt(format_args!("TOCK_DEBUG({}): {}:{}: ", count, file, line));
     let _ = write(writer, args);
+}
+
+pub fn debug_verbose_format(args: Arguments, file_line: &(&'static str, u32)) {
+    let writer = unsafe { get_debug_writer() };
+
+    print_debug_header(writer, args, file_line);
```

Comment:
```suggestion
    print_debug_header(writer, file_line);
    let _ = write(writer, args);
```

Feels like this part should be here (or the "print header" function should be renamed to something like `debug_println_with_header`)

Reply:
Ah yes, I was trying to dedup that extra line, but then the function name gets tricky. I have moved the arg print back into both functions and updated the function names. Thanks!